packages/adapters/src/adapters/Cornerstone/ParametricMap.ts

```diff
@@ -0,0 +1,415 @@
+        const referencedSeriesSequence = Array.isArray(ReferencedSeriesSequence)
+            ? ReferencedSeriesSequence[0]
+            : ReferencedSeriesSequence;
+        const ReferencedSeriesInstanceUID =
+            referencedSeriesSequence.SeriesInstanceUID;
+
+        imageId = getImageIdOfSourceImagebyGeometry(
+            ReferencedSeriesInstanceUID,
+            FrameOfReferenceUID,
+            PerFrameFunctionalGroup,
+            imageIds,
+            metadataProvider,
+            tolerance
+        );
+    }
+
+    return imageId;
+}
+
+function getImageIdOfSourceImageBySourceImageSequence(
+    SourceImageSequence,
+    sopUIDImageIdIndexMap
+) {
+    const { ReferencedSOPInstanceUID, ReferencedFrameNumber } =
+        SourceImageSequence;
+
+    return ReferencedFrameNumber
+        ? getImageIdOfReferencedFrame(
+              ReferencedSOPInstanceUID,
+              ReferencedFrameNumber,
+              sopUIDImageIdIndexMap
+          )
+        : sopUIDImageIdIndexMap[ReferencedSOPInstanceUID];
+}
+
+function getImageIdOfSourceImagebyGeometry(
+    ReferencedSeriesInstanceUID,
+    FrameOfReferenceUID,
+    PerFrameFunctionalGroup,
+    imageIds,
+    metadataProvider,
+    tolerance
+) {
+    if (
+        ReferencedSeriesInstanceUID === undefined ||
+        PerFrameFunctionalGroup.PlanePositionSequence === undefined ||
+        PerFrameFunctionalGroup.PlanePositionSequence[0] === undefined ||
+        PerFrameFunctionalGroup.PlanePositionSequence[0]
+            .ImagePositionPatient === undefined
+    ) {
+        return undefined;
+    }
+
+    for (
+        let imageIdsIndex = 0;
+        imageIdsIndex < imageIds.length;
+        ++imageIdsIndex
+    ) {
+        const sourceImageMetadata = metadataProvider.get(
+            "instance",
+            imageIds[imageIdsIndex]
+        );
+
+        if (
+            sourceImageMetadata === undefined ||
+            sourceImageMetadata.ImagePositionPatient === undefined ||
+            sourceImageMetadata.FrameOfReferenceUID !== FrameOfReferenceUID ||
+            sourceImageMetadata.SeriesInstanceUID !==
+                ReferencedSeriesInstanceUID
+        ) {
+            continue;
+        }
+
+        if (
+            compareArrays(
+                PerFrameFunctionalGroup.PlanePositionSequence[0]
+                    .ImagePositionPatient,
+                sourceImageMetadata.ImagePositionPatient,
+                tolerance
+            )
+        ) {
+            return imageIds[imageIdsIndex];
+        }
+    }
+}
+
+function getImageIdOfReferencedFrame(
+    sopInstanceUid,
+    frameNumber,
+    sopUIDImageIdIndexMap
+) {
+    const imageId = sopUIDImageIdIndexMap[sopInstanceUid];
+
+    if (!imageId) {
+        return;
+    }
+
+    const imageIdFrameNumber = Number(imageId.split("frame=")[1]);
+
+    return imageIdFrameNumber === frameNumber - 1 ? imageId : undefined;
+}
+
+const ParametricMapObj = {
+    generateParametricMap,
```

Comment:
Does this generate parametric maps? Have you tried it? I'm not sure if we planned to implement this.

Reply:
I was working on it but I then stopped because it was not in the scope of this ticket. I forgot to delete that code (done)